emojidex/replace.js

```diff
@@ -1,43 +1,28 @@
+const ZWSP = "​" // ゼロ幅スペース U+200B
+const ZWSP2 = "‌" // ゼロ幅スペース U+2009
 
-window.addEventListener('load', main, false);
 
-function main(e) {
-  const jsInitCheckTimer = setInterval(checkTextare, 1000);
-  let isTextareaExist = false;
-  function checkTextare(){
-    let nowTextareaExist = false;
-    if(document.querySelector('.js-compose-text') != null){
-      nowTextareaExist = true
-    }
-    if(!isTextareaExist && nowTextareaExist){
-      console.log('add function')
-      $('.js-compose-text').bind('input propertychanged', emoji_replace);
-    }
-    isTextareaExist = nowTextareaExist
+document.addEventListener('keydown', event => {
+  const textarea = document.querySelector('.js-compose-text')
+  if (textarea.isConfigured) {
+    return
   }
-}
+  textarea.isConfigured = true
 
-function emoji_replace(){
-  let val = this.value.split(":")
-  for(i in val){
-    if(i % 2){
-      if(i == (val.length - 1)){
-        val[i] = `:${val[i]}`
-        continue;
-      }
-      code = val[i].split(':').join('')
-      if (code != '') {
-        axios
-          .get('https://www.emojidex.com/api/v1/emoji/' + code)
-          .then(response => {
-            if (response.data.moji != null) {
-              this.value = this.value
-                .replace(':' + code + ':', response.data.moji)
-              this.dispatchEvent(new Event('change'));
-            }
-          })
-          .catch(e => {console.log(`emoji replace error:${e}`)})
-      }
-    }
-  }
-}
+  textarea.addEventListener('input', event => {
+    const { target } = event
+
+    const candidates = new Set()
+    target.value = target.value.replace(new RegExp(`:([^: ${ZWSP}${ZWSP2}]+?):`, 'g'), (match, p1) => {
+      candidates.add(p1)
+      return `:${ZWSP}${p1}:${ZWSP2}`
+    })
+
+
+    candidates.forEach(candidate => {
+      chrome.runtime.sendMessage({ code: candidate }, emoji => {
+        target.value = target.value.replace(new RegExp(`:${ZWSP}${candidate}:${ZWSP2}`, 'g'), emoji || `:${candidate}:`)
```

Comment:
```suggestion
        target.value = target.value.replace(new RegExp(`:${ZWSP}${candidate}:${ZWSP2}`, 'g'), emoji || `:${candidate}:`)
        target.dispatchEvent(new Event('change'))
```
`target.dispatchEvent(new Event('change'))`
絵文字を置き換えた後に、tweet内容として反映させるもの。

Reply:
忘れてました ありがとう😉